lib/source/encoding.pm

```diff
@@ -0,0 +1,72 @@
+package source::encoding;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.01';
+
+our $ascii_hint_bits = 0x00000010;
+
+sub import {
+    unimport();     # Get rid of any 'use utf8'
+    my (undef, $arg) = @_;
+    if (lc $arg eq 'utf8') {
+        require utf8;
+        utf8->import;
+        return;
+    }
+    elsif (lc $arg eq 'ascii') {
+        $^H |= $source::encoding::ascii_hint_bits;
```

Comment:
`$ascii_hint_bits` is in scope here (declared via `our`). Why use a fully-qualified name?